lib/openstudio-standards/prototypes/common/objects/Prototype.Model.rb

```diff
@@ -2546,4 +2557,94 @@ def model_add_transfer_air(model)
 
     return true
   end
+
+  # Offset values of a schedule
+  # Main usage is for modeling occupancy standby mode
+  # where the thermostat schedule in these mode are
+  # required to setup/back their thermostat setpoints
+  #
+  # @param time_offset_hash [Hash] Hash providing time (key) and schedule value offset (values)
+  # @param schedule [OpenStudio::model::ScheduleRuleset] OpenStudio schedule object
+  # @return [OpenStudio::model::ScheduleRuleset] Modified OpenStudio schedule object
+  def model_offset_schedule_value(schedule, time_offset_hash)
+    offset_sch = schedule.clone(schedule.model).to_ScheduleRuleset.get
+
+    # Get day schedule
+    day_schedules = []
+    default_day_schedule = offset_sch.defaultDaySchedule
+    day_schedules << default_day_schedule
+    offset_sch.scheduleRules.each do |rule|
+      day_schedules << rule.daySchedule
+    end
+
+    # Offset schedule values
+    day_schedules.each do |day_schedule|
+      (0..23).each do |hr|
+        t = OpenStudio::Time.new(0, hr, 0, 0)
+
+        # Get schedule value
+        value = day_schedule.getValue(t)
+
+        # Offset schedule value
+        if time_offset_hash.key?(hr.to_s)
+          day_schedule.addValue(t, value)
+          t_p_1 = OpenStudio::Time.new(0, hr + 1, 0, 0)
+          day_schedule.addValue(t_p_1, value + time_offset_hash[hr.to_s])
+        end
+      end
+    end
+
+    offset_sch.setName("#{schedule.name} - offset")
+    return offset_sch
+  end
```

Comment:
New method to offset schedule values using a hash that associate the time of the day to the offset value.